### PR TITLE
Make server simple

### PR DESCRIPTION
**Pull request by ductiletoaster**
This is a campaign to simplify the server (and honestly how we report file stats). This was a major pain point for the two mechanisms. We had a high reliance on create react app utilities which I have mostly removed outside of a few more minor functions and plugins.

This work mostly unifies the file reporting (still a bit more to be done but it can wait) and allows us to more easily configure the dev server. 

A few things we need to do next.

- [ ] Make it so we can run a single code path at a time (client only or server only)
- [ ] Create a new build entry for libraries (no need to update dev server)
- [ ] Client only code path needs html web pack plugin so we can server just client code.

### CODE REVIEW

File: tsconfig.json [hunk 1]
```diff
@@ -4,6 +4,7 @@
       "allowSyntheticDefaultImports": true,
       "baseUrl": ".",
       "declaration": true,
+      "downlevelIteration": true,
```
Comment (ductiletoaster) [inline tsconfig.json]: Needed this to support some newer for of loop features with Maps

File: packages/pixeloven-webpack/file-reporter/src/helper.ts [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+    // Transform parsing error into syntax error
+    // TODO: move this to our ESLint formatter?
+    lines = lines.map(line => {
+        const parsingError = /Line (\d+):(?:(\d+):)?\s*Parsing error: (.+)$/.exec(
+            line,
+        );
+        if (!parsingError) {
+            return line;
+        }
+        const [, errorLine, errorColumn, errorMessage] = parsingError;
```
Comment (jimmynotjim) [inline packages/pixeloven-webpack/file-reporter/src/helper.ts]: Why an empty first index?

File: packages/pixeloven-webpack/file-reporter/src/helper.ts [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+                .replace("Error: ", "")
+                .replace(
+                    "Module not found: Cannot find file:",
+                    "Cannot find file:",
+                ),
+        ];
+    }
+
+    // Add helpful message for users trying to use Sass for the first time
+    if (lines[1] && lines[1].match(/Cannot find module.+node-sass/)) {
```
Comment (jimmynotjim) [inline packages/pixeloven-webpack/file-reporter/src/helper.ts]: Should this be `dart-sass` now?

File: packages/pixeloven-webpack/file-reporter/src/helper.ts (hunk 2 above)
Comment (ductiletoaster) [reply packages/pixeloven-webpack/file-reporter/src/helper.ts]: This is a copy paste from create react app. Likely a param they don't need so they just short handed it.

File: packages/pixeloven-webpack/file-reporter/src/helper.ts (hunk 3 above)
Comment (ductiletoaster) [reply packages/pixeloven-webpack/file-reporter/src/helper.ts]: Probably just do both but again this is a copy paste and honestly we probably don't need it. I want to clean this all up but not right now.

**Pull request merged**